File: typescript/impl/RockPlugin.ts
import { BasePlugin } from '../BasePlugin';

export class RockPlugin implements BasePlugin {

    setup(): void{
        console.log("Get ready, setting up...");
    }
    step1(): void{
        console.log("Setting volume and bass for rock!");
    }
    step2(): void{
        console.log("Playing rock");
    }
    teardown(): void{
        console.log("Volume down, stop rock");
    }
}

File: python/impl/rock_plugin.py
from base_plugin import BasePlugin

class RockPlugin(BasePlugin):
    def setup(self):
        print("Get ready, setting up...") 
    
    def step_1(self):
        print("Setting volume and bass for rock!")

    def step_2(self):
        print("Playing rock")

    def teardown(self):
        print("Volume down, stop rock")